examples/lujain_work_area/Contact_Book_OOP/app_user.py

```diff
@@ -0,0 +1,39 @@
+# app_user.py
+
+from contact_book import ContactBook
+
+class AppUser:
+    def __init__(self, username, email):
+        self.username = username
+        self.email = email
+        self.contact_book = ContactBook()
+
+    def add_contact(self, contact):
+        """Add a contact to the user's ContactBook."""
+        self.contact_book.add_contact(contact)
+
+    def delete_contact(self, identifier):
+        """Delete a contact from the user's ContactBook."""
+        self.contact_book.delete_contact(identifier)
+
+    def update_contact(self, identifier, new_contact):
+        """Update a contact in the user's ContactBook."""
+        self.contact_book.update_contact(identifier, new_contact)
+
+    def search_contact(self, identifier):
+        """Search for a contact in the user's ContactBook."""
+        return self.contact_book.search_contact(identifier)
+
+    def save_contact_book(self, filename, format):
+        """Save the user's ContactBook to a file."""
+        if format == 'csv':
```

Comment:
We can avoid this if else, if we use anstract methods and the implementation that is going to be used will be decided dynamically